stingray/io.py

```diff
@@ -653,3 +654,39 @@ def get_btis(gtis, start_time=None, stop_time=None):
 def gti_len(gti):
     """Return the total good time from a list of GTIs."""
     return np.sum([g[1] - g[0] for g in gti])
+
+
+def save_object(object, filename):
+    """
+    Pickle a class instance.
+
+    Parameters
+    ----------
+    object: a class instance
+    filename: str (name of the file to be created)
+
+    """
+
+    directory = os.getcwd() + "/data/"
+    if not os.path.exists(directory):
+        os.makedirs(directory)
+
+    filename = directory + filename
+    pickle.dump(object, open(filename, "wb" ) )
+
+def retrieve_object(filename):
+    """
+    Return a pickled class instance.
+
+    Parameters
+    ----------
+    filename: str (name of the file to be retrieved)
+
+    """
+
+    directory = os.getcwd() + "/data/"
+    if not os.path.exists(directory):
+        os.makedirs(directory)
```

Comment:
1. If you have not committed yet and want to modify the previous commit, you can use `amend`.

2. If you have committed multiple times and want to merge(squash) all of them into one, you can use interactive rebasing. [An example](https://robots.thoughtbot.com/git-interactive-rebase-squash-amend-rewriting-history) to demonstrate.

It is not recommended though, because rebasing can be complicated at times. If you find it comfortable then you should go ahead, otherwise there is no issue.